runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -13255,6 +13256,452 @@ TR::Register *J9::X86::I386::TreeEvaluator::dwrtbariEvaluator(TR::Node *node, TR
+
+            deps->addPostCondition(eaxReg, TR::RealRegister::eax, cg);
+            deps->addPostCondition(edxReg, TR::RealRegister::edx, cg);
+            deps->addPostCondition(ecxReg, TR::RealRegister::ecx, cg);
+            deps->addPostCondition(ebxReg, TR::RealRegister::ebx, cg);
+            deps->addPreCondition(eaxReg, TR::RealRegister::eax, cg);
+            deps->addPreCondition(edxReg, TR::RealRegister::edx, cg);
+            deps->addPreCondition(ecxReg, TR::RealRegister::ecx, cg);
+            deps->addPreCondition(ebxReg, TR::RealRegister::ebx, cg);
+
+            generateRegRegInstruction(TR::InstOpCode::MOV4RegReg, node, ebxReg, valueReg->getLowOrder(), cg); // forming the ECX:EBX pair
+            generateRegRegInstruction(TR::InstOpCode::MOV4RegReg, node, ecxReg, valueReg->getHighOrder(), cg);
+
+            instr = generateRegMemInstruction(TR::InstOpCode::L4RegMem, node, eaxReg, lowMR, cg); // forming the EDX:EAX pair
+            generateRegMemInstruction(TR::InstOpCode::L4RegMem, node, edxReg, highMR, cg);
+            lowMR->setIgnoreVolatile();
+            highMR->setIgnoreVolatile();
+            lowMR->setProcessAsLongVolatileLow();
+            highMR->setProcessAsLongVolatileHigh();
+
+            TR::MemoryReference  *cmpxchgMR = generateX86MemoryReference(node, cg);
+            generateMemInstruction (cg->comp()->target().isSMP() ? TR::InstOpCode::LCMPXCHG8BMem : TR::InstOpCode::CMPXCHG8BMem, node, cmpxchgMR, deps, cg);
+
+            cg->stopUsingRegister(eaxReg);
+            cg->stopUsingRegister(edxReg);
+            cg->stopUsingRegister(ecxReg);
+            cg->stopUsingRegister(ebxReg);
+            }
+         else
+            {
+            instr = generateMemRegInstruction(TR::InstOpCode::S4MemReg, node, lowMR, valueReg->getLowOrder(), cg);
+            generateMemRegInstruction(TR::InstOpCode::S4MemReg, node, highMR, valueReg->getHighOrder(), cg);
+
+            TR::SymbolReference& mrSymRef = lowMR->getSymbolReference();
+            if (mrSymRef.isUnresolved())
+               {
+               padUnresolvedDataReferences(node, mrSymRef, cg);
+               }
+            }
+         }
+      }
+   cg->decReferenceCount(valueChild);
+   if (lowMR && !(valueChild->isDirectMemoryUpdate() && node->getOpCode().isIndirect()))
+      lowMR->decNodeReferenceCounts(cg);
+
+
+   if (node->getSymbolReference()->getSymbol()->isVolatile())
+      {
+      TR_OpaqueMethodBlock *caller = node->getOwningMethod();
+      if ((lowMR || highMR) && caller)
+         {
+         TR_ResolvedMethod *m = comp->fe()->createResolvedMethod(cg->trMemory(), caller, node->getSymbolReference()->getOwningMethod(comp));
+         if (m->getRecognizedMethod() == TR::java_util_concurrent_atomic_AtomicLong_lazySet)
+            {
+            if (lowMR)
+               lowMR->setIgnoreVolatile();
+            if (highMR)
+               highMR->setIgnoreVolatile();
+            }
+         }
+      }
```

Comment:
This could be refactored to be a method in OpenJ9, and a stub method in OMR that does nothing. Can be called something like `lStoreEvaluatorSetHighLowMRIfNeeded` (or a better name along those lines).